### PR TITLE
Fixed maven package gpg issue

### PR DESCRIPTION
**Pull request by Fosol**
Fixed issues with Macos scripts.  They should now work.
Resolved gpg signing issues and Maven Nexus authentication issues for some Docker containers that demand having `sudo` instead of the default `vscode` user.  I would like to solve why this is, but it's far too complex for my understanding.

### CODE REVIEW

File: docs/DEVELOPMENT.md [hunk 1]
```diff
@@ -18,12 +18,14 @@ If you choose to run the solution with Docker you will need to install the follo
...
+| [Git Bash](https://git-scm.com/)                       |                                                         | All scripts in this project are `bash` which natively isn't supported by Windows.       |
+| [chocolately](https://chocolatey.org/install)          |                                                         | Package manager for Windows.                                                            |
+| [make](https://community.chocolatey.org/packages/make) | [make](https://formulae.brew.sh/formula/make)           | Useful tool for speeding up common activities.                                          |
 |                                                        | [coreutils](https://formulae.brew.sh/formula/coreutils) |
+|                                                        | [grep](https://formulae.brew.sh/formula/grep)           | Macos doesn't natively support GNU Grep, so you need to install it. `brew install grep` |
+|                                                        | [gnupg](https://formulae.brew.sh/formula/gnupg)         | Generate signing keys `brew install gpnupg`                                             |
```
Comment (Fosol) [inline docs/DEVELOPMENT.md:28]: Need to create signing keys

File: libs/java/dal/db/.devcontainer/Dockerfile [hunk 2]
```diff
@@ -16,7 +16,7 @@ RUN if [ "${NODE_VERSION}" != "none" ]; then su vscode -c "umask 0002 && . /usr/
 
 # [Optional] Uncomment this section to install additional OS packages.
 RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
-    && apt-get -y install --no-install-recommends build-essential dnsutils
+    && apt-get -y install --no-install-recommends build-essential dnsutils gnupg2
```
Comment (Fosol) [inline libs/java/dal/db/.devcontainer/Dockerfile:19]: not certain if this is needed, but it's supposed to provide a way to link your keys to the OS host.

File: libs/java/dal/db/dal-db/pom.xml [hunk 3]
```diff
@@ -230,6 +230,12 @@
                 <goals>
                   <goal>sign</goal>
                 </goals>
+                <configuration>
```
Comment (Fosol) [inline libs/java/dal/db/dal-db/pom.xml:233]: I need to remove this

File: libs/java/dal/db/scripts/close-staging-repo.sh [hunk 4]
```diff
@@ -7,5 +7,5 @@ if [ ! -z "$varStagingRepoId" ]
 then
   cd dal-db
   export MAVEN_OPTS="--add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.desktop/java.awt.font=ALL-UNNAMED"
-  mvn nexus-staging:close -DstagingRepositoryId="cabcgovtno-$varStagingRepoId"
+  sudo env PATH="$PATH" mvn nexus-staging:close -DstagingRepositoryId="cabcgovtno-$varStagingRepoId" -s /home/vscode/.m2/settings.xml
```
Comment (Fosol) [inline libs/java/dal/db/scripts/close-staging-repo.sh:10]: For some reason we need to use `sudo` to be able to `gpg` when the docker container is hosted on some operating systems (MacOS, WSL).

File: libs/java/dal/db/scripts/setup.sh [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 # This fixes a bug with GPG when signing the package.
-export GPG_TTY=$(tty)
-export MAVEN_OPTS="--add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.desktop/java.awt.font=ALL-UNNAMED"
+# export GPG_TTY=$(tty)
+# export MAVEN_OPTS="--add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.lang.reflect=ALL-UNNAMED --add-opens=java.base/java.text=ALL-UNNAMED --add-opens=java.desktop/java.awt.font=ALL-UNNAMED"
 
+# Import the private.key that is in the root folder.
+# This is required to sign and verify packages.
+# sudo env "PATH=$PATH" gpg --import private.key
+sudo gpg --import private.key
```
Comment (Fosol) [inline libs/java/dal/db/scripts/setup.sh:10]: Required `sudo` to import the key.  For some reason some containers won't work without.  I don't understand why.

File: tools/scripts/gen-maven-files.sh [hunk 6]
```diff
@@ -7,7 +7,7 @@ echo "*********************************"
 
 . ./tools/scripts/variables.sh
 
-ossrhUsername=$(grep -Po '<username>\K[^\<]+' ./libs/java/dal/db/.devcontainer/maven-settings.xml 2>/dev/null)
+ossrhUsername=$("$GREP" -Po '<username>\K[^\<]+' ./libs/java/dal/db/.devcontainer/maven-settings.xml 2>/dev/null)
```
Comment (Fosol) [inline tools/scripts/gen-maven-files.sh:10]: Handle different OS cli option.

File: tools/scripts/os.sh [hunk 7]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+if [[ "$OSTYPE" == "linux-gnu"* ]]; then # Linux
```
Comment (Fosol) [inline tools/scripts/os.sh:3]: Depending on the OS use the appropriate command.  I could have simplified this, but decided to keep all the different options.

**Review by sturufous — APPROVED** [review]

**Pull request merged**